Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs

```diff
@@ -16,46 +16,182 @@ public class UIConfigurationTests : WithAutoFixture
+            var uiModuleCreateNode = uiModuleCustomization.Nodes.FirstOrDefault();
+
+            Assert.NotNull(responseNode);
+            Assert.NotNull(uiModuleCreateNode);
+            Assert.Equal(responseNode.Key, uiModuleCreateNode.Key);
+            Assert.Equal(responseNode.Enabled, uiModuleCreateNode.Enabled);
+        }
+
+        [Fact]
+        public async Task Get_Returns_NotFound_If_Config_Doesnt_Exist()
+        {
+            //Arrange
+            var module = A<string>();
+
+            //Act
+            using var response = await UIConfigurationHelper.SendGetRequestAsync(TestEnvironment.DefaultOrganizationId, module);
+
+            //Assert
+            Assert.Equal(HttpStatusCode.NotFound ,response.StatusCode);
+        }
+
+        [Fact]
+        public async Task Get_Returns_Forbidden_If_Not_Allowed_To_Read()
         {
             //Arrange
             var module = A<string>();
-            var key = "Module.Key";
-            var isEnabled = false;
+            var (cookie, organization) = await CreatePrerequisitesAsync();
+            var (cookieUser2, _) = await CreatePrerequisitesAsync();
+
+            await UIConfigurationHelper.CreateUIModuleAndSaveAsync(organization.Id, module, cookie);
+
+            //Act
+            using var response = await UIConfigurationHelper.SendGetRequestAsync(organization.Id, module, cookieUser2);
+
+            //Assert
+            Assert.Equal(HttpStatusCode.Forbidden ,response.StatusCode);
+        }
 
-            var body = new UIModuleCustomizationDTO
+        [Fact]
+        public async Task Put_Deletes_Nodes()
+        {
+            var module = A<string>();
+            var multipleUiCustomizations= UIConfigurationHelper.PrepareTestUiModuleCustomizationDto(10);
+            var singleUiCustomization = UIConfigurationHelper.PrepareTestUiModuleCustomizationDto();
+            var (cookie, organization) = await CreatePrerequisitesAsync();
+
+            using var firstPutResponse = await UIConfigurationHelper.SendPutRequestAsync(organization.Id, module, multipleUiCustomizations, cookie);
+            Assert.Equal(HttpStatusCode.NoContent, firstPutResponse.StatusCode);
+
+            using var secondPutResponse = await UIConfigurationHelper.SendPutRequestAsync(organization.Id, module, singleUiCustomization, cookie);
+            Assert.Equal(HttpStatusCode.NoContent, secondPutResponse.StatusCode);
+
+            var response = await UIConfigurationHelper.GetCustomizationByModuleAsync(organization.Id, module);
+            Assert.Equal(singleUiCustomization.Nodes.Count(), response.Nodes.Count());
+            Assert.Single(response.Nodes);
+            
+            var responseNode = response.Nodes.FirstOrDefault();
+            var singleUiNode = singleUiCustomization.Nodes.FirstOrDefault();
+
+            Assert.NotNull(responseNode);
+            Assert.NotNull(singleUiNode);
+            Assert.Equal(singleUiNode.Enabled, responseNode.Enabled);
```

Comment:
helper methods for assertion... these blocks are repeated in the other test that verifies results